samples/snippets/quickstart/sub.py

```diff
@@ -42,7 +42,7 @@ def callback(message: pubsub_v1.subscriber.message.Message) -> None:
     try:
         # Calling result() on StreamingPullFuture keeps the main thread from
         # exiting while messages get processed in the callbacks.
-        streaming_pull_future.result(timeout=timeout)
+        streaming_pull_future.result(timeout=timeout and float(timeout))
```

Comment:
Hi Changli, thanks for the bug and the fix. 

I think it'll be simpler if we remove the change to this line and instead add a "type=float" argument to the `add_argument` call below. `Argparse` already supports requiring arguments to be of a certain type. See [docs](https://docs.python.org/3/library/argparse.html#argparse-type).

Reply:
I made a PR to keep this going forward. https://github.com/googleapis/python-pubsub/pull/1051